radix-engine-utils/src/rocks_db_metrics/mod.rs

```diff
@@ -0,0 +1,544 @@
+        root.fill(&WHITE)?;
+        root.margin(20, 20, 20, 20);
+
+        let mut scatter_ctx = ChartBuilder::on(&root)
+            .caption(caption, ("sans-serif", 20).into_font())
+            .x_label_area_size(40)
+            .y_label_area_size(80)
+            .margin(20)
+            .build_cartesian_2d(x_min..x_max, y_min..y_max)?;
+        scatter_ctx
+            .configure_mesh()
+            .x_desc("Size [bytes]")
+            .y_desc("DB read duration [microseconds]")
+            .axis_desc_style(("sans-serif", 16))
+            .draw()?;
+        // 1. draw all read points
+        scatter_ctx
+            .draw_series(
+                data.iter()
+                    .map(|(x, y)| Circle::new((*x, *y), 2, GREEN.filled())),
+            )?
+            .label(format!("Reads (count: {})", data.len()))
+            .legend(|(x, y)| Circle::new((x + 10, y), 2, GREEN.filled()));
+        // 2. draw median for each read series (basaed on same size)
+        scatter_ctx
+            .draw_series(
+                output_data
+                    .iter()
+                    .map(|(x, y)| Cross::new((*x, *y), 6, RED)),
+            )?
+            .label(output_data_name)
+            .legend(|(x, y)| Cross::new((x + 10, y), 6, RED));
+        // 3. draw linear approximetion line
+        scatter_ctx
+            .draw_series(LineSeries::new(
+                lin_x_axis
+                    .values()
+                    .map(|x| (x, (lin_slope * x as f64 + lin_intercept) as f32)),
+                &BLUE,
+            ))?
+            .label(format!(
+                "Linear approx.: f(x)={:.4}*x+{:.1}",
+                lin_slope, lin_intercept
+            ))
+            .legend(|(x, y)| PathElement::new(vec![(x, y), (x + 20, y)], &BLUE));
+        scatter_ctx
+            .configure_series_labels()
+            .background_style(&WHITE)
+            .border_style(&BLACK)
+            .label_font(("sans-serif", 16))
+            .position(SeriesLabelPosition::UpperMiddle)
+            .draw()?;
+
+        root.present().expect("Unable to write result to file");
+
+        // print some informations
+        println!("Read count: {}", data.len());
+        println!(
+            "Distinct size read count: {}",
+            substate_db.read_metrics.borrow().len()
+        );
+        println!(
+            "Read counts list (size, count): {:?}",
+            substate_db
+                .read_metrics
+                .borrow()
+                .iter()
+                .map(|(k, v)| (*k, v.len()))
+                .collect::<Vec<(usize, usize)>>()
+        );
+        println!(
+            "{} points list (size, time): {:?}",
+            output_data_name, output_data
+        );
+        println!(
+            "Linear approx.:  f(size) = {} * size + {}\n",
+            lin_slope, lin_intercept
+        );
+
+        Ok(())
+    }
+
+    pub fn export_graph_and_print_summary_for_two_series(
```

Comment:
Amazing that you've automated linear regression and even graphing!

Reply:
Just using some dedicated crates.